internal/controllers/reconciliation/patch_test.go

```diff
@@ -299,3 +300,164 @@ func TestPatchDeletionBeforeUpgrade(t *testing.T) {
+	})
+
+	// The configmap is deleted after the patch is applied, and it won't be re-created
+	// even with the reconcile interval annotation.
+	err := downstream.Get(ctx, client.ObjectKeyFromObject(cm), cm)
+	require.True(t, errors.IsNotFound(err))
+}
+
+// TestPatchDeletionForResourceWithReconciliationFromInput proves that a patch resource won't be triggered to
+// delete the resource with reconcile-interval it references if the patch with lower readiness group
+func TestPatchDeletionForResourceWithReconciliationFromInput(t *testing.T) {
+	scheme := runtime.NewScheme()
+	corev1.SchemeBuilder.AddToScheme(scheme)
+	testv1.SchemeBuilder.AddToScheme(scheme)
+
+	ctx := testutil.NewContext(t)
+	mgr := testutil.NewManager(t)
+	upstream := mgr.GetClient()
+	downstream := mgr.DownstreamClient
+
+	// Test configmap setup
+	cmName := "test-obj"
+	cmNamespace := "default"
+
+	registerControllers(t, mgr)
+	testutil.WithFakeExecutor(t, mgr, func(ctx context.Context, s *apiv1.Synthesizer, input *krmv1.ResourceList) (*krmv1.ResourceList, error) {
+		obj := &unstructured.Unstructured{
+			Object: map[string]any{
+				"apiVersion": "eno.azure.io/v1",
+				"kind":       "Patch",
+				"metadata": map[string]any{
+					"name":      cmName,
+					"namespace": cmNamespace,
+					"annotations": map[string]string{
+						// This patch should be applied before the configmap is re-created.
+						"eno.azure.io/readiness-group": "1",
+					},
+				},
+				"patch": map[string]any{
+					"apiVersion": "v1",
+					"kind":       "ConfigMap",
+					"ops": []map[string]any{
+						{"op": "add", "path": "/metadata/deletionTimestamp", "value": "2024-01-22T19:13:15Z"},
+					},
+				},
+			},
+		}
+
+		cm := &unstructured.Unstructured{
+			Object: map[string]any{
+				"apiVersion": "v1",
+				"kind":       "ConfigMap",
+				"metadata": map[string]any{
+					"name":      cmName,
+					"namespace": cmNamespace,
+					"annotations": map[string]string{
+						"eno.azure.io/readiness-group":    "2",
+						"eno.azure.io/reconcile-interval": "1ms",
+					},
+				},
+			},
+		}
+
+		return &krmv1.ResourceList{Items: []*unstructured.Unstructured{obj, cm}}, nil
+	})
+
+	// Test subject
+	setupTestSubject(t, mgr)
+	mgr.Start(t)
+
+	// Create deletion patch and configmap.
+	_, comp := writeGenericComposition(t, upstream)
+
+	testutil.Eventually(t, func() bool {
+		err := upstream.Get(ctx, client.ObjectKeyFromObject(comp), comp)
+		return err == nil && comp.Status.CurrentSynthesis != nil && comp.Status.CurrentSynthesis.Reconciled != nil
+	})
+
+	cm := &corev1.ConfigMap{}
+	cm.Name = cmName
+	cm.Namespace = cmNamespace
+	err := downstream.Get(ctx, client.ObjectKeyFromObject(cm), cm)
+	require.NoError(t, err)
+
+	uid := cm.GetUID()
+	// Ensure the configmap is reconciled at least once.
+	time.Sleep(100 * time.Millisecond)
+	err = downstream.Get(ctx, client.ObjectKeyFromObject(cm), cm)
+	require.NoError(t, err)
+
+	newUID := cm.GetUID()
+	// Verify the configmap is not re-created.
```

Comment:
the configmap with higher readiness group will be re-queue and deletion patch is delivered before confimap create or and reconciled, so the configmap won't be deleted. Unless the synthesizer pod create and apply the deletion patch to delete existing configmap.